### ISSUE TITLE
npm shrinkwrap cannot handle https url's, gives unrelated errors

### ISSUE THREAD

**Issue by lukeapage** [npm/npm#8106]
I am getting a specific problem trying to shrinkwrap a repo which references a tarball download of a specific version of jsdoc.

If I specify a older version as a version number, it works, but as a tarball it fails. It seems to be an error using tarballs and shrinkwrap together? any help would be appreciated.

run this in cli to reproduce:
```
mkdir test
cd test
npm init
...
npm install --save-dev https://github.com/jsdoc3/jsdoc/tarball/b6e5051d2fefa49373e964eaa0a508d02203bc87
npm shrinkwrap
```

version 2.8.4

output
```
npm WARN unmet dependency c:\Projects\test\node_modules\jsdoc\node_modules\catharsis\node_modules\underscore-contrib requires underscore@'1.6.0' but will load
npm WARN unmet dependency c:\Projects\test\node_modules\jsdoc\node_modules\underscore,
npm WARN unmet dependency which is version 1.7.0
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "c:\\Program Files (x86)\\nodejs\\node.exe" "c:\\Users\\xlpa\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "shrinkwrap"
npm ERR! node v0.10.30
npm ERR! npm  v2.8.4

npm ERR! Problems were encountered
npm ERR! Please correct and try again.
npm ERR! invalid: underscore@1.7.0 c:\Projects\test\node_modules\jsdoc\node_modules\underscore
npm ERR! extraneous: crypto-browserify@0.1.1 c:\Projects\test\node_modules\jsdoc\node_modules\crypto-browserify
npm ERR! extraneous: tv4@1.0.15 c:\Projects\test\node_modules\jsdoc\node_modules\tv4
npm ERR! extraneous: buffer-browserify@0.2.3 c:\Projects\test\node_modules\jsdoc\node_modules\buffer-browserify
npm ERR! extraneous: url@0.10.2 c:\Projects\test\node_modules\jsdoc\node_modules\url
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     c:\Projects\test\npm-debug.log
```

If npm can't handle url's in some way it should exit early with an error message saying what it cannot handle.

**Comment by lukeapage** [comment]
might be related to #4435

running `npm prune` results in a less severe output

```
npm WARN unmet dependency c:\Projects\test\node_modules\jsdoc\node_modules\catharsis\node_modules\underscore-contrib requires underscore@'1.6.0' but will load
npm WARN unmet dependency c:\Projects\test\node_modules\jsdoc\node_modules\underscore,
npm WARN unmet dependency which is version 1.7.0
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "c:\\Program Files (x86)\\nodejs\\node.exe" "c:\\Users\\xlpa\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "shrinkwrap"
npm ERR! node v0.10.30
npm ERR! npm  v2.8.4

npm ERR! Problems were encountered
npm ERR! Please correct and try again.
npm ERR! invalid: underscore@1.7.0 c:\Projects\test\node_modules\jsdoc\node_modules\underscore
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     c:\Projects\test\npm-debug.log
```

npm debug log is

```
0 info it worked if it ends with ok
1 verbose cli [ 'c:\\Program Files (x86)\\nodejs\\node.exe',
1 verbose cli   'c:\\Users\\xlpa\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'shrinkwrap' ]
2 info using npm@2.8.4
3 info using node@v0.10.30
4 warn unmet dependency c:\Projects\test\node_modules\jsdoc\node_modules\catharsis\node_modules\underscore-contrib requires underscore@'1.6.0' but will load
4 warn unmet dependency c:\Projects\test\node_modules\jsdoc\node_modules\underscore,
4 warn unmet dependency which is version 1.7.0
5 verbose stack Error: Problems were encountered
5 verbose stack Please correct and try again.
5 verbose stack invalid: underscore@1.7.0 c:\Projects\test\node_modules\jsdoc\node_modules\underscore
5 verbose stack     at shrinkwrap_ (c:\Users\xlpa\AppData\Roaming\npm\node_modules\npm\lib\shrinkwrap.js:37:15)
5 verbose stack     at c:\Users\xlpa\AppData\Roaming\npm\node_modules\npm\lib\shrinkwrap.js:31:5
5 verbose stack     at c:\Users\xlpa\AppData\Roaming\npm\node_modules\npm\lib\ls.js:47:30
5 verbose stack     at c:\Users\xlpa\AppData\Roaming\npm\node_modules\npm\node_modules\read-installed\read-installed.js:138:5
5 verbose stack     at c:\Users\xlpa\AppData\Roaming\npm\node_modules\npm\node_modules\read-installed\read-installed.js:251:14
5 verbose stack     at cb (c:\Users\xlpa\AppData\Roaming\npm\node_modules\npm\node_modules\slide\lib\async-map.js:47:24)
5 verbose stack     at c:\Users\xlpa\AppData\Roaming\npm\node_modules\npm\node_modules\read-installed\read-installed.js:251:14
5 verbose stack     at cb (c:\Users\xlpa\AppData\Roaming\npm\node_modules\npm\node_modules\slide\lib\async-map.js:47:24)
5 verbose stack     at c:\Users\xlpa\AppData\Roaming\npm\node_modules\npm\node_modules\read-installed\read-installed.js:251:14
5 verbose stack     at cb (c:\Users\xlpa\AppData\Roaming\npm\node_modules\npm\node_modules\slide\lib\async-map.js:47:24)
6 verbose cwd c:\Projects\test
7 error Windows_NT 6.1.7601
8 error argv "c:\\Program Files (x86)\\nodejs\\node.exe" "c:\\Users\\xlpa\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "shrinkwrap"
9 error node v0.10.30
10 error npm  v2.8.4
11 error Problems were encountered
11 error Please correct and try again.
11 error invalid: underscore@1.7.0 c:\Projects\test\node_modules\jsdoc\node_modules\underscore
12 error If you need help, you may report this error at:
12 error     <https://github.com/npm/npm/issues>
13 verbose exit [ 1, true ]
```

**Comment by lukeapage** [comment]
If I manually go in and create a node_modules with underscore with the right version, shrinkwrap succeeds..

but why doesn't `npm install` do that?

**Comment by lukeapage** [comment]
this has been fixed in npm@3.5.0
